source/includes/fundamentals/code-snippets/IndexPage.java

```diff
@@ -0,0 +1,150 @@
+
+        mongoClient = MongoClients.create(uri);
+        database = mongoClient.getDatabase("sample_mflix");
+        collection = database.getCollection("movies");
+    }
+
+    public static void main(String[] args) {
+        IndexPage page = new IndexPage();
+        page.singleIndex();
+        page.compoundIndex();
+        page.multiKeyIndex();
+        page.textIndex();
+        page.geoSpatialIndex();
+    }
+
+    private void singleIndex() {
+        System.out.println("single index");
+        // begin single index
+        String resultCreateIndex = collection.createIndex(Indexes.ascending("title"));
+        System.out.println(String.format("Index created: %s", resultCreateIndex));
+        // end single index
+
+        // begin covered single query
+        Bson filter = eq("title", "Batman");
+        Bson sort = Sorts.ascending("title");
+        Bson projection = fields(include("title"), excludeId());
+        FindIterable<Document> cursor = collection.find(filter).sort(sort).projection(projection);
+        // end covered single query
+        cursor.forEach(doc -> System.out.println(doc));
+
+    }
+
+    private void compoundIndex() {
+        System.out.println("compound index");
+        // begin compound index
+        String resultCreateIndex = collection.createIndex(Indexes.ascending("type", "rated"));
+        System.out.println(String.format("Index created: %s", resultCreateIndex));
+        // end compound index
+
+        // begin covered compound query
+        Bson filter = and(eq("type", "movie"), eq("rated", "G"));
+        Bson sort = Sorts.ascending("type", "rated");
+        Bson projection = fields(include("type", "rated"), excludeId());
+        FindIterable<Document> cursor = collection.find(filter).sort(sort).projection(projection);
+        // end covered compound query
+        cursor.forEach(doc -> System.out.println(doc));
+    }
+
+    private void multiKeyIndex() {
+        System.out.println("multikey index");
+        // begin multikey index
+        String resultCreateIndex = collection.createIndex(Indexes.ascending("rated", "genres", "title"));
+        System.out.println(String.format("Index created: %s", resultCreateIndex));
+        // end multikey index
+
+        // begin covered multikey query
+        Bson filter = and(eq("genres", "Animation"), eq("rated", "G"));
+        Bson sort = Sorts.ascending("title");
+        Bson projection = fields(include("title", "rated"), excludeId());
+        FindIterable<Document> cursor = collection.find(filter).sort(sort).projection(projection);
+        // end covered multikey query
+        cursor.forEach(doc -> System.out.println(doc));
+    }
+
+    private void textIndex() {
+        System.out.println("text index");
+        // begin text index
+        // create a text index of the "fullplot" field in the "movies" collection
+        // if a text index already exists with a different configuration, this will
+        // error
```

Comment:
This comment brings up a pretty important footgun w.r.t. text indexes -- the fact that you can only have one per collection. The rest of the comment is rendered redundant by the description. Would it make sense to lift this comment out into a warning about how you can only create one text index per collection? Otherwise I think it's unclear why this "will error".

Also, this comment mentions the "fullplot" field but the code itself uses the "plot" field. Which is correct?